controller/program/src/instructions/swap/internal_utils.rs

```diff
@@ -33,27 +28,30 @@ use sanctum_spl_token_jiminy::{
     instructions::transfer::transfer_checked_ix_account_handle_perms,
     sanctum_spl_token_core::{
         instructions::transfer::{NewTransferCheckedIxAccsBuilder, TransferCheckedIxData},
-        state::{
-            account::{RawTokenAccount, TokenAccount},
-            mint::{Mint, RawMint},
-        },
+        state::mint::{Mint, RawMint},
     },
 };
 
 use crate::{
-    pricing::{NewPpIxPreAccsBuilder, PriceExactInIxAccountHandles},
-    svc::{lst_sync_sol_val_unchecked, NewSvcIxPreAccsBuilder, SvcIxAccountHandles},
+    svc::lst_sync_sol_val_unchecked,
     verify::{verify_not_rebalancing_and_not_disabled, verify_pks},
     Cpi,
 };
 
-#[inline]
-pub fn process_swap_exact_in(
-    abr: &mut Abr,
-    accounts: &[AccountHandle<'_>],
+pub type SwapIxAccounts<'a, 'acc> = SwapIxAccs<
+    AccountHandle<'acc>,
+    SwapIxPreAccountHandles<'acc>,
+    &'a [AccountHandle<'acc>],
+    &'a [AccountHandle<'acc>],
+    &'a [AccountHandle<'acc>],
+>;
+
+pub fn swap_checked<'a, 'acc>(
+    abr: &Abr,
+    accounts: &'a [AccountHandle<'acc>],
     args: &IxArgs,
-    cpi: &mut Cpi,
-) -> Result<(), ProgramError> {
+    expected_accs_builder: NewIxPreAccsBuilder<&[u8; 32]>,
```

Comment:
why is this passed in as an arg?

Reply:
yaa, i thought of this and i think the reasoning I had was what if tomorrow `NewSwapExactInIxPreAccsBuilder` is different from `NewSwapExactOutIxPreAccsBuilder` or so? 

but i think we can keep it the same as just `NewIxPreAccsBuilder` for now